sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -191,22 +196,56 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
     cancelledDate: Date,
     newIssuingProvince: string,
   ): Promise<UpdateResult> {
-    if (!cancelledDate || !newIssuingProvince) {
-      throw new Error(
-        "Not all required fields to update a received MSFAA record were provided.",
+    return this.dataSource.transaction(async () => {
+      // Keeping both MSFAA Cancellation update and saving of MSFAA Notification message as a part of the same transaction.
+      if (!cancelledDate || !newIssuingProvince) {
```

Comment:
do we need this check inside transaction?

Reply:
Moved it outside. Thanks 👍